src/schemes/boundary/open_boundary/system.jl

```diff
@@ -15,184 +12,132 @@ Open boundary system for in- and outflow particles.
 
     fluid_system_index = Ref(0)
 
     smoothing_length = initial_smoothing_length(fluid_system)
 
-    boundary_candidates = fill(false, nparticles(initial_condition))
+    boundary_candidates = fill(false, nparticles(initial_conditions))
     fluid_candidates = fill(false, nparticles(fluid_system))
 
-    return OpenBoundarySPHSystem(boundary_model, initial_condition, fluid_system,
+    boundary_zone_indices = zeros(UInt8, nparticles(initial_conditions))
+
+    boundary_zones_new = map(zone -> BoundaryZone(zone.initial_condition,
+                                                  zone.spanning_set,
+                                                  zone.zone_origin,
+                                                  zone.zone_width,
+                                                  zone.flow_direction,
+                                                  zone.plane_normal,
+                                                  nothing,
+                                                  zone.average_inflow_velocity,
+                                                  zone.prescribed_density,
+                                                  zone.prescribed_pressure,
+                                                  zone.prescribed_velocity),
+                             boundary_zones)
+
+    return OpenBoundarySPHSystem(boundary_model, initial_conditions, fluid_system,
                                  fluid_system_index, smoothing_length, mass, density,
                                  volume, pressure, boundary_candidates, fluid_candidates,
-                                 boundary_zone, reference_velocity_,
-                                 reference_pressure_, reference_density_, buffer, cache)
+                                 boundary_zone_indices, boundary_zones_new, buffer, cache)
 end
 
-function create_cache_open_boundary(boundary_model, initial_condition,
-                                    reference_density, reference_velocity,
-                                    reference_pressure)
-    ELTYPE = eltype(initial_condition)
+function initialize!(system::OpenBoundarySPHSystem, semi)
+    (; boundary_zones) = system
 
-    prescribed_pressure = isnothing(reference_pressure) ? false : true
-    prescribed_velocity = isnothing(reference_velocity) ? false : true
-    prescribed_density = isnothing(reference_density) ? false : true
+    update_boundary_zone_indices!(system, initial_coordinates(system), boundary_zones, semi)
 
-    if boundary_model isa BoundaryModelTafuni
-        return (; prescribed_pressure=prescribed_pressure,
-                prescribed_density=prescribed_density,
-                prescribed_velocity=prescribed_velocity)
-    end
+    return system
+end
+
+function create_cache_open_boundary(boundary_model, initial_condition, reference_values)
+    ELTYPE = eltype(initial_condition)
+
+    pressure_references = map(ref -> ref.reference_pressure, reference_values)
```

Comment:
```suggestion
    # Separate `reference_values` into pressure, density and velocity reference values
    pressure_references = map(ref -> ref.reference_pressure, reference_values)
```
Also, I'm not sure about the name "references". It sounds weird to me. But `density_reference_values` is a bit long. Do you have any other ideas?

Reply:
No, I don't. They call it also "reference values" in the literature...

Reply:
I'm usually against abbreviations, but here I think I would prefer `pressure_ref_values` over `pressure_references`.
Or even `pressure_reference_values` if the long name is not too tedious.